plugins/repository-gcs/src/main/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageBlobStore.java

```diff
@@ -259,19 +226,27 @@ void writeBlob(String blobName, InputStream inputStream, long blobSize) throws I
         for(int retry = 0; retry < 3; retry++){
             
                 try { 
+                    logger.debug("WriteBlob() try " + (retry+1));
                     SocketAccess.doPrivilegedIOException( () -> {
+                        logger.debug("Trying to write blob " + blobName);
                         InputStreamContent stream = new InputStreamContent(null, inputStream);
+                        logger.debug("Opened input stream");
                         stream.setLength(blobSize);
-                        
+                        logger.debug("Set blob size");
                         Storage.Objects.Insert insert = client.objects().insert(bucket, null, stream);
+                        logger.debug("Create insert operation");
                         insert.setName(blobName);
+                        insert.getMediaHttpUploader().setDirectUploadEnabled(false);
+                        logger.debug("Set blob name in the operation and also set resumable upload");
                         insert.execute();
+                        logger.debug("Executed blob insertion for" + blobName);
                         return null;
                     });
+                    return;
                 }catch(Exception e){
                     
                         logger.warn("[repository-gcs][writeBlob] Lost - Throttling try #" + retry );
-                        throw e;
+                        logger.warn(e);
```

Comment:
I think we may need to throw an exception on the last retry - otherwise nobody will know if the blob fails to get written